src/proxy/NearCachedMapProxy.ts

```diff
@@ -40,9 +40,13 @@ export class NearCachedMapProxy<K, V> extends MapProxy<K, V> {
 
         this.nearCache = this.client.getNearCacheManager().getOrCreateNearCache(name);
         if (this.nearCache.isInvalidatedOnChange()) {
+            const logger = this.client.getLoggingService().getLogger();
             this.addNearCacheInvalidationListener().then((id) => {
                 this.invalidationListenerId = id;
                 this.nearCache.setReady();
+            }).catch((e) => {
+                logger.warn('NearCachedMapProxy', 'Failed to register Near Cache invalidation listener for '
+                    + name + ' map.', e);
```

Comment:
If we have an error here, should we reject the `ready` promise? If we don't do that, the next call to `get` will not be resolved forever since it waits for the `ready` promise. 

By the way, should the other methods in this proxy also wait for that promise?

Reply:
> If we have an error here, should we reject the `ready` promise? If we don't do that, the next call to `get` will not be resolved forever since it waits for the `ready` promise.

Good idea. Addressed in 1f24513adcb8c2ce67f355ae9a7478a6cc968923

> By the way, should the other methods in this proxy also wait for that promise?

As far as I understand, this deferred promise is required to postpone reads until the listener is ready, so that the chances that near cache is properly warmed up are higher. Writes in near cache are local as it's a write-behind cache and that's probably why the promise is ignored there.

In general, I'd prefer to wait for the listener in `client.getMap()` rather than in each individual operation. Such behavior should be less error-prone. Created #684 to track this enhancement.

Reply:
Yes, makes sense